community/values/src/main/java/org/neo4j/values/VirtualValue.java

```diff
@@ -33,6 +33,10 @@
     @Override
     public final boolean equals( Object other )
     {
+        if ( (other instanceof SequenceValue) && (this instanceof SequenceValue) )
```

Comment:
the whole goal of new values API was to remove casts as much as possible from all the equality methods, this change does not follow that paradigm (this place and all other updated equals methods) is there is any way we can avoid that?